src/components/@shared/TokenApproval/index.tsx

```diff
@@ -29,34 +30,39 @@ export default function TokenApproval({
       : ddo.services[0].datatokenAddress
   const spender = price.address
 
-  const checkTokenApproval = useCallback(async () => {
-    // if (!tokenAddress || !spender || !isAssetNetwork || !amount) return
-    // const allowance = await ocean.datatokens.allowance(
-    //   tokenAddress,
-    //   accountId,
-    //   spender
-    // )
-    // amount &&
-    //   new Decimal(amount).greaterThan(new Decimal('0')) &&
-    //   setTokenApproved(
-    //     new Decimal(allowance).greaterThanOrEqualTo(new Decimal(amount))
-    //   )
-  }, [tokenAddress, spender, accountId, amount, isAssetNetwork])
+  // TODO: how to check if token is approved as .allowance does not exist?
```

Comment:
Ideas for checking if token and amount have already been approved for user + pool are very welcome here.

Reply:
fixed in new version of ocean.js